### ISSUE TITLE
`.serializeArray()` evaluates the attribute instead of the property.

### ISSUE THREAD

**Issue by anseki** [cheeriojs/cheerio#1364]
Hello, thank you for the great library.

`.serializeArray()` seems to evaluate each value from `.attr()` instead of a value from `.prop()`.
That is, it gets status from only default HTML source.
For example:

```html
<form id="frm1">
  <input type="checkbox" value="check_on" name="chk1" id="chk1">
  <input type="checkbox" value="check_on" name="chk2" id="chk2" checked>
  <input type="checkbox" value="check_on" name="chk3" id="chk3">
</form>
```

```js
const $form = $('#frm1');

console.log('======== Default status');
console.log('#chk1: %s', $('#chk1', $form).prop('checked'));
console.log('#chk2: %s', $('#chk2', $form).prop('checked'));
console.log('#chk3: %s', $('#chk3', $form).prop('checked'));
console.dir($form.serializeArray()); // expected result

console.log('======== Change `prop`');
$('#chk1', $form).prop('checked', true);
$('#chk2', $form).prop('checked', false);
$('#chk3', $form).prop('checked', true);

console.log('#chk1: %s', $('#chk1', $form).prop('checked'));
console.log('#chk2: %s', $('#chk2', $form).prop('checked'));
console.log('#chk3: %s', $('#chk3', $form).prop('checked'));
console.dir($form.serializeArray()); // not expected result

// Workaround
console.log('======== Change `prop` and `attr`');
$('#chk1', $form).prop('checked', true).attr('checked', 'checked');
$('#chk2', $form).prop('checked', false).removeAttr('checked');
$('#chk3', $form).prop('checked', true).attr('checked', 'checked');

console.log('#chk1: %s', $('#chk1', $form).prop('checked'));
console.log('#chk2: %s', $('#chk2', $form).prop('checked'));
console.log('#chk3: %s', $('#chk3', $form).prop('checked'));
console.dir($form.serializeArray()); // expected result
```

Result:

```
======== Default status
#chk1: false
#chk2: true
#chk3: false
[ { name: 'chk2', value: 'check_on' } ]
======== Change `prop`
#chk1: true
#chk2: false
#chk3: true
[ { name: 'chk2', value: 'check_on' } ]
======== Change `prop` and `attr`
#chk1: true
#chk2: false
#chk3: true
[ { name: 'chk1', value: 'check_on' },
  { name: 'chk3', value: 'check_on' } ]
```

The second `.serializeArray()` returns an array as the same as first one, but `checked` of checkboxes were already changed.
That returns correct array if attribute was changed via `.attr()` instead of `.prop()`.

Also, `.serializeArray()` of jQuery returns correct array. That is, the second `.serializeArray()` after changing via `.prop()` returns `[ { name: 'chk1', value: 'check_on' }, { name: 'chk3', value: 'check_on' } ]`.

**Comment by fb55** [comment]
Turns out that this is an issue with `prop`. A bug fix is pending as #1579.

**Comment by anseki** [comment]
Thank you. :smile:
